src/hr/dto/create-hr.dto.ts

```diff
@@ -0,0 +1,10 @@
+import { IsString } from 'class-validator';
+import { HrEntity } from 'src/types';
+
+export class CreateHrDto implements Omit<HrEntity, 'id' | 'user' | 'maxReservedStudents'> {
```

Comment:
Warto stworzyć oddzielny interface w stylu `HrCreateRequest` w którym to wszystko opiszesz. Można wtedy wykorzystać go na froncie - pamiętaj, że na froncie przy takim dzieleniu typów nie możemy wykorzystać `CreateHrDto` bo wyskoczą błędy
Spójrz analogicznie do typów `student.request.ts`

Reply:
Nie widzę czegoś takiego jak student.request.ts w typach.